core/ostctools.c

```diff
@@ -57,25 +57,37 @@ void ostctools_import(const char *file, struct dive_table *divetable)
 	// Read dive number from the log
 	uc_tmp = calloc(2, 1);
 	fseek(archive, 258, 0);
-	fread(uc_tmp, 1, 2, archive);
+	if (fread(uc_tmp, 1, 2, archive) != 2) {
+		report_error(translate("gettextFromC", "Failed to read '%s'"), file);
+		free(uc_tmp);
+		goto out;
+	}
 	ostcdive->number = uc_tmp[0] + (uc_tmp[1] << 8);
 	free(uc_tmp);
 
 	// Read device's serial number
 	uc_tmp = calloc(2, 1);
 	fseek(archive, 265, 0);
-	fread(uc_tmp, 1, 2, archive);
+	if (fread(uc_tmp, 1, 2, archive) != 2) {
+		report_error(translate("gettextFromC", "Failed to read '%s'"), file);
+		free(uc_tmp);
+		goto out;
+	}
 	serial = uc_tmp[0] + (uc_tmp[1] << 8);
 	free(uc_tmp);
 
 	// Read dive's raw data, header + profile
 	fseek(archive, 456, 0);
-	while (!feof(archive)) {
-		fread(buffer + i, 1, 1, archive);
+	while ((c = getc(archive)) != EOF) {
+		buffer[i] = c;
```

Comment:
i would probably add a sentence about the change to `getc()` from `fread()` in the commit message.